packages/storefront-sdk/src/client/index.ts

```diff
@@ -12,8 +23,16 @@ export class StorefrontClient {
 		});
 	}
 
-	// NEW METHODS GO HERE :)
-	placeholder() {
-		return this.#graphqlClient.query('', {}).toPromise();
+	query<QData = any, QVariables extends AnyVariables = any>({
+		query,
+		variables
+	}: {
+		query: TypedDocumentNode<QData, QVariables> | DocumentNode | string;
+		variables?: QVariables | string;
+	}): Promise<StorefrontResponse<QData>> {
+		return this.#graphqlClient
+			.query(query, variables as QVariables)
+			.toPromise()
+			.then((res) => ({ data: res.data, error: res.error }));
```

Comment:
Not a problem at all, but we could use destructuring here:

```suggestion
			.then(({ data, error }) => ({ data, error }));
```